zrml/swaps/src/tests.rs

```diff
@@ -908,152 +930,123 @@ fn pool_exit_subsidy_fails_if_pool_does_not_exist() {
 #[test]
 fn pool_exit_subsidy_fails_if_scoring_rule_is_not_rikiddo() {
     ExtBuilder::default().build().execute_with(|| {
-        create_initial_pool_with_funds_for_alice(ScoringRule::CPMM, true);
+        create_initial_pool_with_funds_for_alice(ScoringRule::CPMM, Some(0), true);
         assert_noop!(
             Swaps::pool_exit_subsidy(alice_signed(), 0, _1),
             crate::Error::<Runtime>::InvalidScoringRule
         );
     });
 }
 
-#[test]
-fn pool_exit_with_exact_pool_amount_exchanges_correct_values() {
+#[test_case(49_999_999_665, 12_272_234_300, 0, 0; "no_fees")]
+#[test_case(45_082_061_850, 12_272_234_300, _1_10, 0; "with_exit_fees")]
+#[test_case(46_403_174_924, 11_820_024_200, 0, _1_20; "with_swap_fees")]
+#[test_case(41_836_235_739, 11_820_024_200, _1_10, _1_20; "with_both_fees")]
```

Comment:
Yeah, very important question. You're right about the reason, it's because we're now charging swap fees. I'm not going to go into the reasoning _why_ we're charging swap fees on `pool_join_*` and `pool_exit_*` functions (there's a reason, I promise).

The test is structured in the following way: Bob creates a pool, Alice joins the pool by calling `pool_join_with_exact_asset_amount` and then exits the pool with the amount of tokens she receives. In an ideal world, we'd get rid of the `pool_join_with_exact_asset_amount` call. The only reason that exists is because Alice needs pool tokens. The test parameter specifies how many pool tokens Alice receives. Since we're charging swap fees in the third and fourth test, the amount of tokens Alice receives is smaller than before (we're swapping the exact amount in).

There are many better ways to do it. For example, we could just deposit pool tokens on Alice's account. But I didn't feel like changing too much of the test logic at this time.